airflow/providers/hashicorp/hooks/vault.py

```diff
@@ -0,0 +1,281 @@
+        * approle: password -> secret_id
+        * aws_iam: login -> key_id, password -> secret_id
+        * azure: login -> client_id, password -> client_secret
+        * ldap: login -> username,   password -> password
+        * userpass: login -> username, password -> password
+        * radius: password -> radius_secret
+
+    :param vault_conn_id: id of the connection to use
+    :type vault_conn_id: str
+    :param auth_type: type of authentication. One of
+    :type auth_type: str
+    :param mount_point: The "path" the secret engine was mounted on. (Default: ``secret``)
+    :type mount_point: str
+    :param kv_engine_version: Select the version of the engine to run (``1`` or ``2``, default: ``2``)
+    :type kv_engine_version: int
+    :param token: Authentication token to include in requests sent to Vault.
+        (for ``token`` and ``github`` auth_type)
+    :type token: str
+    :param role_id: Role (for ``aws_iam``, 'approle', auth_types)
+    :type role_id: str
+    :param kubernetes_role: Role for Authentication (for ``kubernetes`` auth_type)
+    :type kubernetes_role: str
+    :param kubernetes_jwt_path: Path for kubernetes jwt token (for ``kubernetes`` auth_type, default:
+        ``/var/run/secrets/kubernetes.io/serviceaccount/token``)
+    :type kubernetes_jwt_path: str
+    :param gcp_key_path: Path to GCP Credential JSON file (for ``gcp`` auth_type)
+    :type gcp_key_path: str
+    :param gcp_keyfile_dict: Dictionary of keyfile parameters. (for ``gcp`` auth_type).
+           Mutually exclusive with gcp_key_path
+    :type gcp_keyfile_dict: dict
+    :param gcp_scopes: Comma-separated string containing GCP scopes (for ``gcp`` auth_type)
+    :type gcp_scopes: str
+    :param azure_tenant_id: Tenant of azure (for ``azure`` auth_type)
+    :type azure_tenant_id: str
+    :param azure_resource: Resource if of azure (for ``azure`` auth_type)
+    :type azure_resource: str
+    :param radius_host: Host for radius (for ``radius`` auth_type)
+    :type radius_host: str
+    :param radius_port: Port for radius (for ``radius`` auth_type)
+    :type radius_port: int
+
+    """
+    def __init__(self,  # pylint: disable=too-many-arguments, too-many-branches
+                 vault_conn_id=None,
+                 auth_type=None,
+                 mount_point=None,
+                 kv_engine_version: Optional[int] = None,
+                 token=None,
+                 role_id=None,
+                 kubernetes_role: Optional[str] = None,
+                 kubernetes_jwt_path=None,
+                 gcp_key_path=None,
+                 gcp_keyfile_dict=None,
+                 gcp_scopes=None,
+                 azure_tenant_id: Optional[str] = None,
+                 azure_resource: Optional[str] = None,
+                 radius_host: Optional[str] = None,
+                 radius_port: Optional[int] = None):
+        super().__init__()
+        connection = self.get_connection(vault_conn_id)
```

Comment:
```suggestion
        self.connection = self.get_connection(vault_conn_id)
```

Reply:
Ah yeah. I actually made a small refactor and split the init into smaller functions, which made it much more readable and modular. Having connection as field helped with that indeed.